### PR TITLE
2.5.2

### PR DESCRIPTION
**Pull request by Scraayp**
Everyone is allowed to review and provide comments!!!!

To Do:
- [x] Recheck if every comment and description is written correctly!
- [x] Add unmute function
- [x] Add after time unmute option
- [x] Muterole and modlogchannel able to be found with id

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -349,7 +349,21 @@ class AntiSpamClient extends EventEmitter {
...
+			}
+			if (this.options.errorMessages) {
+				await message.channel
+					.send(this.format(this.options.muteErrorMessage, message))
+					.catch((e) => {
+						if (this.options.verbose) {
+							console.log(`DAntiSpam (muteUser#muteRoleNotFound): ${e.message}`)
+						}
+					})
+			}
```
Comment (Androz2091) [inline index.js:365]: Here you should add `return false` so the rest of the code won't be run.

**Review by Androz2091 — DISMISSED** [review]

File: index.js (hunk 1 above)
Comment (Scraayp) [reply index.js:365]: Added

Comment (Androz2091) [reply index.js:365]: Not here. Above, so it returns if the role does not exist.

File: index.d.ts [hunk 2]
```diff
@@ -81,7 +81,7 @@ declare module 'discord-anti-spam' {
 		kickEnabled?: boolean;
 		banEnabled?: boolean;
 		muteEnabled?: boolean;
-		muteRoleName?: String;
+		muteRole?: String;
 		modLogsChannelName?: string;
```
Comment (Scraayp) [inline index.d.ts:85]: Needs to be done too.

File: index.js (hunk 1 above)
Comment (Scraayp) [reply index.js:365]: It is on the if(!role)

Comment (Scraayp) [reply index.js:365]: image
look

Comment (Scraayp) [reply index.js:365]: nvm fixed it in the last commit.

**Comment by Scraayp** [comment]
Added todo at the top!

**Comment by Scraayp** [comment]
Maybe there is an idea to make the warn/mute/kick/ban function public to allow people to create their own commands.

**Comment by Scraayp** [comment]
Added unmute abbility, unmute after time and added the option for unmute after time

**Review by adonyssantos — APPROVED** [review]

**Comment by eightless** [comment]
make multiple mutes and bans and kicks, and prob warns

**Comment by Scraayp** [comment]
Ye, I have tested it out. The full version is broken, I will work on it.

**Comment by Scraayp** [comment]
All of these will be added in the 2.6.0 update

**Pull request closed without merging**